subsys/bluetooth/services/ots.c

```diff
@@ -0,0 +1,281 @@
+
+	LOG_DBG("OTS Object Name GATT Read Operation");
+
+	if (!ots->cur_obj) {
+		LOG_ERR("No Current Object selected in OTS!");
+		return BT_GATT_ERR(BT_GATT_OTS_OBJECT_NOT_SELECTED);
+	}
+
+	return bt_gatt_attr_read(conn, attr, buf, len, offset,
+				 ots->cur_obj->metadata.name,
+				 strlen(ots->cur_obj->metadata.name) + 1);
+}
+
+static ssize_t ots_obj_type_read(struct bt_conn *conn,
+				 const struct bt_gatt_attr *attr, void *buf,
+				 u16_t len, u16_t offset)
+{
+	struct bt_gatt_ots *ots = (struct bt_gatt_ots *) attr->user_data;
+	struct bt_gatt_ots_obj_metadata *obj_meta;
+
+	LOG_DBG("OTS Object Type GATT Read Operation");
+
+	if (!ots->cur_obj) {
+		LOG_ERR("No Current Object selected in OTS!");
+		return BT_GATT_ERR(BT_GATT_OTS_OBJECT_NOT_SELECTED);
+	}
+
+	obj_meta = &ots->cur_obj->metadata;
+	if (obj_meta->type.uuid == BT_UUID_TYPE_128) {
+		return bt_gatt_attr_read(conn, attr, buf, len, offset,
+					 obj_meta->type.uuid_128,
+					 sizeof(obj_meta->type.uuid_128));
+	} else {
+		return bt_gatt_attr_read(conn, attr, buf, len, offset,
+					 &obj_meta->type.uuid_16,
+					 sizeof(obj_meta->type.uuid_16));
+	}
+}
+
+static ssize_t ots_obj_size_read(struct bt_conn *conn,
+		const struct bt_gatt_attr *attr, void *buf,
+		u16_t len, u16_t offset)
+{
+	struct bt_gatt_ots *ots = (struct bt_gatt_ots *) attr->user_data;
+
+	LOG_DBG("OTS Object Size GATT Read Operation");
+
+	if (!ots->cur_obj) {
+		LOG_ERR("No Current Object selected in OTS!");
+		return BT_GATT_ERR(BT_GATT_OTS_OBJECT_NOT_SELECTED);
+	}
+
+	return bt_gatt_attr_read(conn, attr, buf, len, offset,
+				 &ots->cur_obj->metadata.size,
+				 sizeof(ots->cur_obj->metadata.size));
+}
+
+static ssize_t ots_obj_id_read(struct bt_conn *conn,
+			       const struct bt_gatt_attr *attr, void *buf,
+			       u16_t len, u16_t offset)
+{
+	struct bt_gatt_ots *ots = (struct bt_gatt_ots *) attr->user_data;
+	struct bt_gatt_ots_obj_id id;
+
+	LOG_DBG("OTS Object ID GATT Read Operation");
+
+	if (!ots->cur_obj) {
+		LOG_ERR("No Current Object selected in OTS!");
+		return BT_GATT_ERR(BT_GATT_OTS_OBJECT_NOT_SELECTED);
+	}
+
+	id.val = ots->cur_obj->metadata.id;
+	LOG_HEXDUMP_DBG(id.ba, sizeof(id.ba), "Current Object ID:");
+
+	return bt_gatt_attr_read(conn, attr, buf, len, offset, id.ba,
+				 sizeof(id.ba));
+}
+
+static ssize_t ots_obj_prop_read(struct bt_conn *conn,
+				 const struct bt_gatt_attr *attr, void *buf,
+				 u16_t len, u16_t offset)
+{
+	struct bt_gatt_ots *ots = (struct bt_gatt_ots *) attr->user_data;
+
+	LOG_DBG("OTS Object Properties GATT Read Operation");
+
+	if (!ots->cur_obj) {
+		LOG_ERR("No Current Object selected in OTS!");
+		return BT_GATT_ERR(BT_GATT_OTS_OBJECT_NOT_SELECTED);
+	}
+
+	return bt_gatt_attr_read(conn, attr, buf, len, offset,
+				 &ots->cur_obj->metadata.props,
+				 sizeof(ots->cur_obj->metadata.props));
+}
+
+
+BT_GATT_SERVICE_DEFINE(ots_svc,
+#if CONFIG_BT_GATT_OTS_SECONDARY_SVC
+	BT_GATT_SECONDARY_SERVICE(BT_UUID_OTS),
```

Comment:
It would end up in a section so theoretically, the other service module that includes OTS could parse the relevant section to extract attribute settings for OTS. However, this is not an ideal solution. Would you like me to add an API for obtaining service descriptor? Or do you have a different solution in mind?